### ISSUE TITLE
add this wiki for newcomers as first line of defense :)

### ISSUE THREAD

**Issue by denfromufa** [pythonnet/pythonnet#158]
https://github.com/denfromufa/pythonnet/wiki/Troubleshooting-FAQ

**Comment by denfromufa** [comment]
@tonyroberts can you please review and add this to readme.md?

**Comment by denfromufa** [comment]
@filmor @tonyroberts @vmuriart what do you guys think about opening up the wiki in pythonnet repository, so that this document above and other similar docs can be added without the need to do pull requests?

If not I can open up a pull request instead.

**Comment by vmuriart** [comment]
Do you mean to add the link, or open the wiki option on the group repo? I think the latter would make more sense. 

I just noticed that [PyOhio](https://www.youtube.com/watch?v=ivTIGR4YhpM) had a talk on pythonnet. I havent checked it out yet, but it might be worth adding.

**Comment by vmuriart** [comment]
I just re-read your email and realized you meant to open up the wiki.... sorry brain hasn't been working today. 
I agree, would be a good place to add documentation and help docs. 

On a separate note, anyone seen this other .net project? https://bitbucket.org/pydotnet/pydotnet/wiki/Home

**Comment by denfromufa** [comment]
@vmuriart thanks for pointing to pydotnet, and no I have not seen this before! Looks like you have some experience with it :)

In summary pydotnet is based on C++/CLI - which is mixed-mode native-managed code that uses special syntax, such as handles:

https://bitbucket.org/pydotnet/pydotnet/src/b01b5bb06cd582c4c29b0d6733fb26fa8c93f206/src/ConvertToManagedObject.cpp?at=default&fileviewer=file-view-default#ConvertToManagedObject.cpp-342

I tried writing C++/CLI few years ago to wrap CPython from .NET and the code leaked badly. So I failed, but these guys seem to manage it.

Anyway pydotnet does not include embedding. No Mono/Linux support is provided and likely will not, due to reverse pinvoke, which is not available in Mono yet.

On the plus note the documentation looks nice and a lot of logic is implemented in pure python for those more familiar with Python than C#. 
But the core (C++/CLI and Boost.Python) is very magical code.

**Comment by denfromufa** [comment]
I opened wiki and added few links and articles:

https://github.com/pythonnet/pythonnet/wiki

**Comment by denfromufa** [comment]
@vmuriart I mentioned that talk on mailing list, the video from youtube is not very good quality, so I requested the presentation from Hussein in email.

**Comment by vmuriart** [comment]
@denfromufa there's a mailing list?

Do you have a copy of your old implementation of CLI wrapper? Just curious to take a look

**Comment by denfromufa** [comment]
@vmuriart I still have the old C++/CLI wrapper and got permission to publish here, but just did not have enough time/incentive.

Here is a mailing list:

https://mail.python.org/pipermail/pythondotnet/

You can post message to it using this email address:

pythondotnet@python.org

And subscribe on this page:

https://mail.python.org/mailman/listinfo/pythondotnet

**Comment by denfromufa** [comment]
@vmuriart if you would like to view the pythonnet PyOhIO presentation from @Ijwu (Hussein Farran), then request from him on LinkedIn. He sent me the presentation and it has nice things like building REPL and using pythonnet with ASP.NET. However I do not have permission to share it here.

**Comment by Ijwu** [comment]
@denfromufa I think I saw your email but forgot to reply to it. But, as you saw, I tried to get right back to you on LinkedIn. Thanks for the interest. :)

@vmuriart I'm open to distributing my slides. The most helpful resource in writing the presentation, other than the source code, is the wiki on gh-pages. (http://pythonnet.github.io/). 

I will spend some time to get the related code for that presentation put together into a repo and published to github. Let me know if there are any requests/anything I can do to help.
